mutable_buffer/src/chunk.rs

```diff
@@ -287,6 +314,58 @@ impl Chunk {
             .collect()
     }
 
+    /// If the column names that matches the predicate can be found
+    /// from the predicate entirely using metadata, returns those
+    /// strings.
+    ///
+    /// If the predicate can not be evaluated entirely with
+    /// metadata, returns Ok(None)
+    pub fn column_names(
+        &self,
+        table_name: &str,
+        chunk_predicate: &ChunkPredicate,
+    ) -> Result<Option<BTreeSet<String>>> {
+        // No support for general purpose expressions
+        if !chunk_predicate.chunk_exprs.is_empty() {
+            return Ok(None);
+        }
+
+        let table_name_id = self.dictionary.lookup_value(table_name).context(
+            InternalTableNotFoundInDictionary {
+                table_name,
+                chunk_id: self.id(),
+            },
+        )?;
+
+        let mut chunk_column_ids = BTreeSet::new();
+
+        // Is this table in the chunk?
+        if let Some(table) = self.tables.values().find(|table| table.id == table_name_id) {
```

Comment:
Is there a reason this is iterating through `values` rather than using the `HashMap` lookup?

```suggestion
        if let Some(table) = self.tables.get(&table_name_id) {
```

Reply:
@carols10cents I assumed it was because the hashmap key id is not the same as `table.id` but is some generic "string value" id, which could be used instead of the same string in other places, e.g., column names or whatnot.

I also spent some time thinking about this iteration though. It's called quite a few times (at least once per column lookup during the operation). But I'm thinking it's going to be negligible because there probably won't be many tables in the chunk.

Reply:
I think it will be possible to lookup the id first -- I'll give it a shot